File: CS-Examples/CS-Examples/24_Workbook/EncryptWorkbook.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace EncryptWorkbook
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook and load a file
            Workbook workbook = new Workbook();
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\EncryptWorkbook.xlsx");

            //Protect Workbook with the password you want
            workbook.Protect("eiceblue");

            //Save the document and launch it
            workbook.SaveToFile("EncryptWorkbook_result.xlsx", ExcelVersion.Version2010);
            ExcelDocViewer("EncryptWorkbook_result.xlsx");
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VB-Examples/24_Workbook/EncryptWorkbook.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace EncryptWorkbook
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the workbook from the specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\EncryptWorkbook.xlsx")

            ' Protect the workbook with a password
            workbook.Protect("eiceblue")

            ' Save the protected workbook to a new file with Excel Version 2010 format
            workbook.SaveToFile("EncryptWorkbook_result.xlsx", ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer("EncryptWorkbook_result.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
